Code formatted

Latency opened the pull request:
Allman style!

tonerdo commented on src/ReadLine/Abstractions/IConsole.cs:
```diff
@@ -1,14 +1,14 @@
-        void Write(string value);
-        void WriteLine(string value);
-    }
+  public interface IConsole
```
This shouldn't be public. Any particular reason you made it so?

tonerdo commented on src/ReadLine/ReadLine.csproj:
```diff
@@ -3,32 +3,57 @@
-    <Authors>Toni Solarin-Sodara</Authors>
-    <TargetFrameworks>netstandard1.3;net45</TargetFrameworks>
-    <DebugType>portable</DebugType>
+    <Authors>Toni Solarin-Sodara;Latency McLaughlin</Authors>
```
Go ahead and remove this line. I'd let GitHub's contributors page handle this

tonerdo commented on src/ReadLine/ReadLine.csproj:
```diff
@@ -3,32 +3,57 @@
   <PropertyGroup>
```
I updated this file on master, you should do a rebase

Latency replied:
Not showing that I need to rebase.   I pulled everything but there was nothing to pull / merge that conflicted with mine which 'was' technically correct.   Now it seems to not sync with compiling after submission.

tonerdo commented on test/ReadLine.Tests/Abstractions/Console2.cs:
```diff
@@ -1,50 +1,45 @@
+      CursorLeft = 0;
+      CursorTop = 0;
+      BufferWidth = 100;
+      BufferHeight = 100;
```
Let's leave the convention of putting `_` before private members and let's keep those private members and allow the public properties to return them

tonerdo commented on test/ReadLine.Tests/ReadLine.Tests.csproj:
```diff
@@ -1,24 +1,22 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
```
This file has also been updated on master

Latency commented:
@tonerdo requested changes on this pull request.

Thanks for the contrib. A couple comments, also let's keep the indention at 4 spaces

  _____  

In src/ReadLine/Abstractions/IConsole.cs <https://github.com/tonerdo/readline/pull/35#discussion_r143343697> :

>  {
-    internal interface IConsole
-    {
-        int CursorLeft { get; }
-        int CursorTop { get; }
-        int BufferWidth { get; }
-        int BufferHeight { get; }
-        void SetCursorPosition(int left, int top);
-        void SetBufferSize(int width, int height);
-        void Write(string value);
-        void WriteLine(string value);
-    }
+  public interface IConsole

This shouldn't be public. Any particular reason you made it so?

 

I don’t believe I made it public for any specific reason other than my profiling tools probably telling me to do so or doing it for me.


Severity

Code

Description

Project

File

Line

Suppression State


Error

CS0051

Inconsistent accessibility: parameter type 'IConsole' is less accessible than method 'KeyHandler.KeyHandler(IConsole, List<string>, Func<string, int, string[]>)'

ReadLine(net47), ReadLine(netcoreapp2.0), ReadLine(netstandard2.0)

D:\Documents\Visual Studio 2017\Projects\ReadLine\src\ReadLine\KeyHandler.cs

22

Active

 

 

  _____  

In src/ReadLine/ReadLine.csproj <https://github.com/tonerdo/readline/pull/35#discussion_r143343710> :

> @@ -3,32 +3,57 @@
   <PropertyGroup>
     <Description>A GNU-Readline like library for .NET/.NET Core</Description>
     <AssemblyTitle>ReadLine</AssemblyTitle>
-    <VersionPrefix>1.2.0</VersionPrefix>
-    <Authors>Toni Solarin-Sodara</Authors>
-    <TargetFrameworks>netstandard1.3;net45</TargetFrameworks>
-    <DebugType>portable</DebugType>
+    <Authors>Toni Solarin-Sodara;Latency McLaughlin</Authors>

Go ahead and remove this line. I'd let GitHub's contributors page handle this

 

> Reverted

  _____  

In src/ReadLine/ReadLine.csproj <https://github.com/tonerdo/readline/pull/35#discussion_r143343721> :

> @@ -3,32 +3,57 @@
   <PropertyGroup>

I updated this file on master, you should do a rebase

> I would have to cherry pick that.  I know for a fact this changes for .NET Core.   Here is what I have:

  <PropertyGroup>

    <Description>A GNU-Readline like library for .NET/.NET Core</Description>

    <AssemblyTitle>ReadLine</AssemblyTitle>

    <Authors>Toni Solarin-Sodara</Authors>

    <Owners>Toni Solarin-Sodara</Owners>

    <TargetFrameworks>net47;netcoreapp2.0;netstandard2.0</TargetFrameworks>

    <Version>1.2.1</Version>

    <AssemblyName>ReadLine</AssemblyName>

    <PackageId>ReadLine</PackageId>

    <PackageTags>readline;gnu;console;shell;cui</PackageTags>

    <PackageProjectUrl>https://github.com/tsolarin/readline</PackageProjectUrl>

    <PackageLicenseUrl>https://github.com/tsolarin/readline/blob/master/LICENSE</PackageLicenseUrl>

    <RepositoryType>git</RepositoryType>

    <RepositoryUrl>https://github.com/tsolarin/readline</RepositoryUrl>

    <GenerateAssemblyTitleAttribute>false</GenerateAssemblyTitleAttribute>

    <GenerateAssemblyDescriptionAttribute>false</GenerateAssemblyDescriptionAttribute>

    <GenerateAssemblyFileVersionAttribute>false</GenerateAssemblyFileVersionAttribute>

    <GenerateAssemblyVersionAttribute>false</GenerateAssemblyVersionAttribute>

    <GenerateAssemblyInformationalVersionAttribute>false</GenerateAssemblyInformationalVersionAttribute>

    <SignAssembly>true</SignAssembly>

    <AssemblyOriginatorKeyFile>snKey.snk</AssemblyOriginatorKeyFile>

    <GeneratePackageOnBuild>true</GeneratePackageOnBuild>

    <PackageReleaseNotes>Updated to support .NET Core 2.0.

Updated to support .NET Standard v2.0.

Updated to support .NET Framework v4.7.

Consolidated redundant code and expressions.

Repacked for multi-frameworks.

Updated versioning.

Updated test projects.</PackageReleaseNotes>

    <Company>Toni Solarin-Sodara</Company>

  </PropertyGroup>

 

  _____  

In test/ReadLine.Tests/Abstractions/Console2.cs <https://github.com/tonerdo/readline/pull/35#discussion_r143343743> :

> -            _cursorTop = top;
-        }
-
-        public void Write(string value)
-        {
-            _cursorLeft += value.Length;
-        }
-
-        public void WriteLine(string value)
-        {
-            _cursorLeft += value.Length;
-        }
+      CursorLeft = 0;
+      CursorTop = 0;
+      BufferWidth = 100;
+      BufferHeight = 100;

Let's leave the convention of putting _ before private members and let's keep those private members and allow the public properties to return them

> Previously, it was incorrect according to default .NET C# / IDE standards.    Because the properties are public, it will dictate being CamelCased as such.  Since they have been changed to an expression-bodied function members which eliminate backing fields; thus, there are no private backing fields and the other option is auto-property  which doesn’t do much justice either.

> Did you explore the latest of C# 6.0? It brought another new feature called “Expression-bodied function members” which will allow you to write expression bodies for properties, methods etc. instead of writing statement blocks. Thus reducing the code.

  _____  

In test/ReadLine.Tests/ReadLine.Tests.csproj <https://github.com/tonerdo/readline/pull/35#discussion_r143343751> :

> @@ -1,24 +1,22 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">

This file has also been updated on master

>  I did notice that at the time.  It is line #1.    Should be okay! 

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub <https://github.com/tonerdo/readline/pull/35#pullrequestreview-67846115> , or mute the thread <https://github.com/notifications/unsubscribe-auth/AAwHSF3XDzhyc4f_xuAmIbnVaxIDr8Q7ks5sqBvsgaJpZM4PxhG2> .  <https://github.com/notifications/beacon/AAwHSI2d-0B6csYThgMYaIPFTwPLsYPFks5sqBvsgaJpZM4PxhG2.gif>

Latency commented:
Can't resolve conflicts.   They look good on my end.  Update master HEAD to match.
